ruby/wipe-appwrite-bucket/index.rb

```diff
@@ -0,0 +1,62 @@
+require 'appwrite'
+
+def main(req, res)
+  client = Appwrite::Client.new
+  storage = Appwrite::Storage.new(client)
+
+  if !req.variables['APPWRITE_FUNCTION_ENDPOINT'] or !req.variables['APPWRITE_FUNCTION_API_KEY']
+    return res.json({
+      :success => false,
+      :message => 'Environment variables are not set. Function cannot use Appwrite SDK.',
+    })
+  end
+
+  bucket_id = nil
+  begin
+    payload = JSON.parse(req.payload)
+    bucket_id = (payload['bucketId'] || '').delete(' ')
+  rescue Exception => err
+    return res.json({
+      :success => false,
+      :message => 'Payload is invalid.',
+    })
+  end
+
+  if !bucket_id || bucket_id == ''
+    return res.json({
+      :success => false,
+      :message => 'bucketId was not provided.',
+    })
+  end
+
+  client
+    .set_endpoint(req.variables['APPWRITE_FUNCTION_ENDPOINT'])
+    .set_project(req.variables['APPWRITE_FUNCTION_PROJECTID'])
+    .set_key(req.variables['APPWRITE_FUNCTION_API_KEY'])
+    .set_self_signed(true)
+
+  done = false
+  while !done do
+    file_list = nil
+    begin
+      file_list = storage.list_files(bucket_id: bucket_id)
+    rescue Exception => err
+      return res.json({
+        :success => false,
+        :message => 'Failed to get file list.',
```

Comment:
I'm getting an error:

```json
{"response":{"success":false,"message":"Failed to get file list."},"stdout":""}
```

despite passing the correct project id, endpoint, and API key. Is this still working for you?

Reply:
I was using a different variable name "APPWRITE_FUNCTION_PROJECTID" instead of "APPWRITE_FUNCTION_PROJECT_ID", just fixed it and retested, works on my end.

Reply:
I tried again but I'm still getting the same error. I used the same variable values in the ruby playground and that worked.